server/http_test.go

```diff
@@ -332,6 +332,125 @@ func TestHandleAuthFuncResponsesMultipleRedirectURLs(t *testing.T) {
 	}
 }
 
+func TestHandleTokenFunc(t *testing.T) {
+
+	fx, err := makeTestFixtures()
+	if err != nil {
+		t.Errorf("could not run test fixtures")
+	}
+
+	tests := []struct {
+		query    url.Values
+		creds    string
+		wantCode int
+	}{
+		// bad grant_type
+		{
+			query: url.Values{
+				"grant_type": []string{"invalid!"},
+				"state":      []string{""},
```

Comment:
Is "state" part of the token request? https://tools.ietf.org/html/rfc6749#section-4.1.3